core/src/test/scala/unit/kafka/coordinator/AbstractCoordinatorConcurrencyTest.scala

```diff
@@ -168,8 +168,37 @@ object AbstractCoordinatorConcurrencyTest {
       watchKeys = Collections.newSetFromMap(new ConcurrentHashMap[TopicPartitionOperationKey, java.lang.Boolean]()).asScala
     }
 
+    override def maybeStartTransactionVerificationForPartition(
+      topicPartition: TopicPartition,
+      transactionalId: String,
+      producerId: Long,
+      producerEpoch: Short,
+      baseSequence: Int,
+      requestLocal: RequestLocal,
+      callback: (Errors, RequestLocal, VerificationGuard) => Unit
+    ): Unit = {
+      // Skip verification
+      callback(Errors.NONE, requestLocal, VerificationGuard.SENTINEL)
+    }
+
     override def tryCompleteActions(): Unit = watchKeys.map(producePurgatory.checkAndComplete)
 
+    override def appendForGroup(timeout: Long,
+                                requiredAcks: Short,
+                                internalTopicsAllowed: Boolean,
+                                origin: AppendOrigin,
+                                entriesPerPartition: Map[TopicPartition, MemoryRecords],
+                                responseCallback: Map[TopicPartition, PartitionResponse] => Unit,
+                                delayedProduceLock: Option[Lock] = None,
+                                recordValidationStatsCallback: Map[TopicPartition, RecordValidationStats] => Unit = _ => (),
+                                requestLocal: RequestLocal = RequestLocal.NoCaching,
+                                actionQueue: ActionQueue = null,
+                                verificationGuards: Map[TopicPartition, VerificationGuard] = Map.empty,
```

Comment:
What is the difference between passing no verification guard and passing `VerificationGuard.SENTINEL`?